File: ExampleCodes/Javascript/Linear.js
console.log("Starting...");

const Linear = (n) => {
  //O(n) complexity
  for (let i = 0; i < n; i++) {
    console.log("i is " + i);
  }
};

const { performance } = require("perf_hooks");
const time1 = performance.now();
Linear(5); //this is n
const time2 = performance.now();
console.log(`Elapsed time: ${(time2 - time1) / 1000} seconds`);

File: ExampleCodes/rust/linear/src/linear.rs
use std::time::Instant;

pub fn linear() {
    println!("Starting ... ");
    let n = 8;

    let start = Instant::now(); //records this instant

    //Linear Time Complexity Example
    for x in 0..n {
        //basically a single plain old loop
        println!("{}", x);
    }

    let finish = Instant::now();

    print!("Took ");
    print!("{:?}", finish.checked_duration_since(start));
    print!(" To Complete");
}
